packages/web/hooks/use-keyboard-navigation.ts

```diff
@@ -0,0 +1,114 @@
+import { RefObject, useCallback } from "react";
+import { useLatest } from "react-use";
+
+import { useConst } from "./use-const";
+import { useKeyActions } from "./use-key-actions";
+import { useStateRef } from "./use-state-ref";
+
+function getItemId(uniqueId: string, index: number) {
+  return `item-selector-item-${uniqueId}-${index}`;
+}
+
+function getElement(
+  dataAttributeName: string,
+  uniqueId: string,
+  index: number
+) {
+  return document.querySelector(
+    `[${dataAttributeName}=${getItemId(uniqueId, index)}]`
+  );
+}
+
+function getAllElements(dataAttributeName: string) {
+  return document.querySelectorAll(`[${dataAttributeName}]`);
+}
+
+/** Navigate a list of items, with an optional search bar to accompany the selection. */
+export function useKeyboardNavigation<TItem>(
+  items: TItem[],
+  onSelectItem: (item: TItem) => void,
+  searchBoxRef: RefObject<HTMLInputElement> | undefined,
+  dataAttributeName = "key-nav-item"
+) {
```

Comment:
Thoughts on making this a parameter object? It makes it easier to read and extend in the future.